### PR TITLE
#13 fix handling of android biometric dialog cancelling

### LINKED ISSUE #13: Android and Ios checkBiometryAuthentication has different behavior
ios returns false if user cancel biometry authentication
android throw an exception if user cancel biometry authentication

### LINKED ISSUE #18: iosSimulatorArm64 support

### LINKED ISSUE #19: Update moko-resources dependency without cinterop-pluralizedString
In `moko-resources` 0.21.0 release cinterop was removed as unused. But Kotlin/Native have own list of dependencies inside klib. All libraries, that depends on moko-resources, have inside own `manifest` file in `klib` dependency to `dev.icerock.moko:resources-cinterop-pluralizedString`. So gradle download new version of moko-resources (0.21.0) and try to compile project, but Kotlin/Native see own dependencies list and see that `moko-biometry` depends on `dev.icerock.moko:resources-cinterop-pluralizedString` but that library not exist anymore and gradle not download it.
As result we see:
```
error: could not find "dev.icerock.moko:resources-cinterop-pluralizedString" in [/Users/amikhailov/.konan/kotlin-native-prebuilt-macos-aarch64-1.8.10/bin, /Users/amikhailov/.konan/klib, /Users/amikhailov/.konan/kotlin-native-prebuilt-macos-aarch64-1.8.10/klib/common, /Users/amikhailov/.konan/kotlin-native-prebuilt-macos-aarch64-1.8.10/klib/platform/ios_arm64]
```

need to publish new version with updated moko resources

### PR DESCRIPTION
**Pull request by Tetraquark**

### CODE REVIEW

File: biometry/src/androidMain/kotlin/dev/icerock/moko/biometry/BiometryAuthenticator.kt [hunk 1]
```diff
@@ -132,26 +132,26 @@ actual class BiometryAuthenticator actual constructor() {
...
 
                     override fun onAuthenticationSucceeded(
-                        result: BiometricPrompt.AuthenticationResult) {
+                        result: BiometricPrompt.AuthenticationResult
+                    ) {
                         super.onAuthenticationSucceeded(result)
                         callback.invoke(Result.success(true))
                     }
-
-                    override fun onAuthenticationFailed() {
```
Comment (Alex009) [inline biometry/src/androidMain/kotlin/dev/icerock/moko/biometry/BiometryAuthenticator.kt:151]: why this removed?

**Comment by Alex009** [comment]
to fix CI need update kotlin version

File: biometry/src/androidMain/kotlin/dev/icerock/moko/biometry/BiometryAuthenticator.kt (hunk 1 above)
Comment (Tetraquark) [reply biometry/src/androidMain/kotlin/dev/icerock/moko/biometry/BiometryAuthenticator.kt:151]: At the documentation: 
> onAuthenticationFailed()  Called when a biometric is valid but not recognized. 

Is it the expected behavior of `checkBiometricAuthentication` when function terminates and returns the result when a fingerprint is incorrect? As if the user has only one attempt.

Comment (Alex009) [reply biometry/src/androidMain/kotlin/dev/icerock/moko/biometry/BiometryAuthenticator.kt:151]: if auth dialog will be dissmissed after `onAuthenticationFailed` then we should call callback. did you check this case?

Comment (Tetraquark) [reply biometry/src/androidMain/kotlin/dev/icerock/moko/biometry/BiometryAuthenticator.kt:151]: After dismissing a dialog callback `onAuthenticationError` is called, not `onAuthenticationFailed`. Callback `onAuthenticationFailed` is called when a fingerprint is incorrect, for example, when user put wrong finger. After incorrect fingerprint the biometric dialog doesn't dismiss automatically.

Comment (Alex009) [reply biometry/src/androidMain/kotlin/dev/icerock/moko/biometry/BiometryAuthenticator.kt:151]: In this case you are right - we should not pass result to coroutines-test at failed call

**Review by Alex009 — CHANGES_REQUESTED** [review]
Ci fix required

**Review by Alex009 — APPROVED** [review]

**Pull request merged**